src/utils/pgwire/src/pg_message.rs

```diff
@@ -289,10 +312,35 @@ impl FeMessage {
             b'H' => Ok(FeMessage::Flush),
             _ => Err(std::io::Error::new(
                 ErrorKind::InvalidInput,
-                format!("Unsupported tag of regular message: {}", val),
+                format!("Unsupported tag of regular message: {}", tag),
             )),
         }
     }
+
+    pub async fn skip_body(
+        stream: &mut (impl AsyncRead + Unpin),
+        header: FeMessageHeader,
+    ) -> Result<()> {
+        let FeMessageHeader {
+            tag: _,
+            payload_len,
+        } = header;
+
+        if payload_len > 0 {
+            // Use smaller batches to process the payload instead of handling it all at once to minimize memory usage.
+            const BUF_SIZE: usize = 1024;
+            let mut buf: Vec<u8> = vec![0; BUF_SIZE];
+            for _ in 0..(payload_len as usize) / BUF_SIZE {
+                stream.read_exact(&mut buf).await?;
+            }
+            let remain = (payload_len as usize) % BUF_SIZE;
+            if remain > 0 {
+                buf.truncate(remain);
+                stream.read_exact(&mut buf).await?;
+            }
```

Comment:
I think skipping with zero allocation can be achieved with a combination of [`tokio::io::copy`](https://docs.rs/tokio/latest/tokio/io/fn.copy.html) and [`tokio::io::sink`](https://docs.rs/tokio/latest/tokio/io/fn.sink.html).

Reply:
> A heap-allocated copy buffer with 8 KB is created to take data from the reader to the writer

tokio::io::copy still need a copy buffer.